File: src/main/java/com/github/morningzeng/toolset/ui/coding/Base64Component.java
package com.github.morningzeng.toolset.ui.coding;

import com.intellij.openapi.project.Project;

import java.util.Base64;
import java.util.function.Supplier;

import static java.nio.charset.StandardCharsets.UTF_8;

/**
 * @author Morning Zeng
 * @since 2024-05-21
 */
public final class Base64Component extends AbstractCodingComponent {

    public Base64Component(final Project project) {
        super(project);
        this.encodeArea.setPlaceholder("Base64 encoded text");
        this.decodeArea.setPlaceholder("Base64 decoded text");
    }

    @Override
    Supplier<String> encode() {
        return () -> Base64.getEncoder().encodeToString(this.decodeArea.getText().getBytes(UTF_8));
    }

    @Override
    Supplier<String> decode() {
        return () -> new String(Base64.getDecoder().decode(this.encodeArea.getText()), UTF_8);
    }

}

File: src/main/kotlin/com/github/morningzeng/toolbox/ui/view/coding/Base64Component.kt
package com.github.morningzeng.toolbox.ui.view.coding

import com.intellij.openapi.project.Project
import java.nio.charset.StandardCharsets
import java.util.*

/**
 * @author Morning Zeng
 * @since 2025-07-22
 */
class Base64Component(
    project: Project
) : AbstractCodingComponent(project) {
    override fun encode(): String {
        return Base64.getEncoder().encodeToString(decodeArea.text.toByteArray(StandardCharsets.UTF_8))
    }

    override fun decode(): String {
        return String(Base64.getDecoder().decode(encodeArea.text), StandardCharsets.UTF_8)
    }
}